### DISCUSSION TITLE
Is there a way to tell the language before recognition

### DISCUSSION THREAD

**Discussion by tungle** [openai/whisper#989] [Q&A]
Hi, 
Is there a way to send to the model a short sound file and it returns the language spoken the majority of the time? 
From my testing, if I know what lang is spoken and pass in the value of the 'lang' for the whole conversation (say, between 2 people), then the result will be better.

**Comment by glangford** [comment]
See this section of the readme with the code sample showing the use of `detect_language`.

> `detect_language()` : "Detect the spoken languages in the audio, and return them as list of strings, along with the ids
    of the most probable language tokens and the probability distribution over all language tokens."

https://github.com/openai/whisper#python-usage
